### DISCUSSION TITLE
v3.2.0

### DISCUSSION THREAD

**Discussion by flottsam** [r-lidar/lidR#468] [General]
Is it possible to download v3.2.0 from here? 
I'd really like to try the voxel rendering as shown in the lidR book.

**Comment by Jean-Romain** [comment]
Yes you can but it is still in development so new stuff can still be modified without warning

```r
remotes::install_github("Jean-Romain/lidR", ref = "devel")
```

**Reply by flottsam** [reply]
Great! Yeah I really just want to check out the voxel rendering shown in the lidR book for some figures haha.

**Reply by Jean-Romain** [reply]
But read the doc first

> `voxel` boolean or numeric. Displays voxels instead of points. Useful to render the output of `voxelize_points` for example. However it is computationally demanding to render and can easily takes 15 seconds for 10000 voxels. It should be reserved to small scenes. If boolean the voxel resolution is guessed automatically. Otherwise user can provide the size of the voxels. An internal optimization get rid of voxels that are not visible when surrounded by other voxels to reduce the rendering time.

**Reply by flottsam** [reply]
Thanks for tip. Great package BTW!

**Reply by flottsam** [reply]
I'm not able to replicate the voxel rendering as shown in the lidR book using the mixedConifer.laz system file with lidR 3.2.0. Is it intended to work with RGL or the lidR Viewer? Any suggestions?

**Comment by Jean-Romain** [comment]
I fixed a bug in 21410ad
This now works
```r
LASfile <- system.file("extdata", "Megaplot.laz", package="lidR")
las <- readLAS(LASfile)
voxels <- voxel_metrics(las, ~mean(Intensity), 8)
plot(voxels, voxel = TRUE)
```

**Comment by flottsam** [comment]
Thanks for the fix, it's great to visualize the actual voxel space. 
I just noticed that the color fades when panning the voxel-cloud. E.g., when rotating the scene for a birds-eye view (and other angles too), the color ramp fades to mostly white (as if simulating some light source).
It's not a big deal, I just wanted to bring it to your attention.

**Reply by Jean-Romain** [reply]
I guess you are talking about that. https://stackoverflow.com/questions/25753975/r-reducing-glare-with-rgl-package

That is certainly an improvement I can add

**Reply by flottsam** [reply]
Yes, glare is the word I was looking for. Thanks again.

**Reply by Jean-Romain** [reply]
I removed specular reflections. It's much better indeed.
